test/functorch/test_aotdispatch.py

```diff
@@ -1070,6 +1070,42 @@ def forward(self, x, y):
             """At compilation time, graph 1 was compiled under the assumption that input 1 would be a duplicate of input 0, but at runtime this was not the case.  This indicates a guard bug in AOTAutograd or Dynamo, please file a bug to PyTorch."""  # noqa: B950
         )
 
+    @patch('functorch._src.aot_autograd.AOT_COUNTER', new_callable=itertools.count)
+    @patch("functorch._src.config.debug_assert", True)
+    def test_invalid_requires_grad_fake(self, counter):
+        shape_env = ShapeEnv()
+        fake_mode = FakeTensorMode(shape_env=shape_env)
+
+        class F(torch.nn.Module):
+            def forward(self, x, y):
+                return (x + y,)
+
+        x = torch.randn(3, 3, requires_grad=True)
+        y = torch.randn(3, 3, requires_grad=True)
+        z = torch.randn(3, 3, requires_grad=False)
+
+        fake_x = fake_mode.from_tensor(x)
+        fake_y = fake_mode.from_tensor(y)
+        fake_z = fake_mode.from_tensor(z)
+
+        # Non-mutating please!
+        def compare(m1, m2, inps):
+            r1, g1 = _outs_and_grads(m1, inps, inps)
+            r2, g2 = _outs_and_grads(m2, inps, inps)
+            self.assertEqual(r1, r2)
+            self.assertEqual(g1, g2)
```

Comment:
can this get factored out?

Reply:
(because iiuc, this is a copy paste of an existing comparison)

Reply:
...in fact... can this entire function get deduplicated with the original `test_invalid_requires_grad`? As far as I can tell, all you have done is replace the inputs xyz with fake inputs right?

Reply:
> can this get factored out?

Sure

> ...in fact... can this entire function get deduplicated with the original test_invalid_requires_grad? As far as I can tell, all you have done is replace the inputs xyz with fake inputs right?

I dislike a lot of helper fn calls in tests, it seems more readable to just have this test as it is, but yes, its identical except w/ fake tensors